src/psyclone/parse/module_info.py

```diff
@@ -92,6 +103,44 @@ def __init__(self, name, filename):
         # each module, indexed by the module names: Dict[str, Set(str)].
         self._used_symbols_from_module = None
 
+        # This variable will be a set that stores the name of all routines
+        # (based on fparser), # so we can test is a routine is defined
```

Comment:
"...can test whether a routine is defined"

Reply:
Given that the costly bit is the creation of the fparser tree, could we not just create PSyIR and examine the resulting symbol table(s)? Or are you after the extra bit of robustness/coverage gained by only relying on fparser2 and not having to convert to PSyIR?

Reply:
Yes, though I am not using it atm. But I have a problem with current LFRic, where log_mod.F90 contains a symbol that is only used in a write statement, i.e. a code block. So my code can't find if this variable is read or written.  In theory, we could use fparser to access these details (though for now it's much easier to work around). I am happy to convert this to PSyIR instead.

Reply:
That's a todo for now till we have support for interfaces.

Reply:
Please could you add a TODO for #2435

Reply:
Done.